src/refract/JSONSchemaVisitor.cc

```diff
@@ -33,6 +33,45 @@ namespace refract
             }
     }
 
+
+    template <typename T>
+    void IncludeMembers(const T& element, typename T::ValueType& members)
+    {
+        const typename T::ValueType* val = GetValue<T>(element);
+
+        if (!val) {
+            return;
+        }
+
+        for (typename T::ValueType::const_iterator it = val->begin();
+             it != val->end();
+             ++it) {
+
+            if (!(*it) || (*it)->empty()) {
+                continue;
+            }
+
+            if ((*it)->element() == "ref") {
+                IElement::MemberElementCollection::const_iterator found = (*it)->attributes.find("resolved");
+
+                if (found == (*it)->attributes.end()) {
+                    continue;
+                }
+
+                const T* resolved = TypeQueryVisitor::as<T>((*found)->value.second);
+
+                if (!resolved) {
+                    throw refract::LogicError("Mixin must refer to same type as parent");
```

Comment:
this should be probable catched one level higher and rethrow it with sourcemap info